core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -123,6 +123,9 @@ private TableProperties() {
   public static final String SPLIT_OPEN_FILE_COST = "read.split.open-file-cost";
   public static final long SPLIT_OPEN_FILE_COST_DEFAULT = 4 * 1024 * 1024; // 4MB
 
+  public static final String READ_DELETE_FILES_WORKER_POOL_SIZE = "read.deletes.num-threads";
+  public static final int READ_DELETE_FILES_WORKER_POOL_SIZE_DEFAULT = 1;  // read delete files in serial.
```

Comment:
I don't think this makes much sense as a table property. Table properties are for table configuration, but this is an engine concern. Many engines handle parallelism internally so this wouldn't be appropriate. I think that Flink should expose a setting and manage the thread pool for using a delete file reader pool.

Reply:
I admit that this configuration may not be appropriate as a table property. But I think this optimization should apply to all engines, not only flink. Such as I will sync mysql cdc data to iceberg by flink and rewrite and merge the delete files into data files by spark. Maybe we chould config this through `SystemProperties` like we config `iceberg.worker.num-threads`.
https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/SystemProperties.java#L35

Reply:
I move this config to `SystemProperties`. So we can stop propagate the 'tableProperties' and handle the parallelism internally.

Reply:
Is it not possible to have Flink pass in the executor service it chooses to use?